src/batSplay.ml

```diff
@@ -314,6 +314,70 @@ struct
     try find k m
     with Not_found -> def
 
+  (* option 1: simple implementation, dummy "find" invocation to perform rebalancing *)
+  let find_first_opt f (map : 'a t) =
+    let rec loop found tr =
+      match tr with
+      | Node(l, (k, v), r)
+        -> if f k
+           then loop (Some (k, v)) l
+           else loop found r
+      | Empty -> found in
+    let ret = loop None (sget map) in
+    (* dummy find to rebalance the tree *)
+    if Option.is_some ret
+    then ignore(find (fst (Option.get ret)) map);
```

Comment:
`Option.get` guarded by `Option.is_some` is not so nice. This would be nicer with `Option.iter` or a direct pattern-matching.

Reply:
after re-writing, the find_first_opt method will call the find_first method, where the rebalancing happens after the "raise Not_found", so the Option.is_some guard is obsolete.